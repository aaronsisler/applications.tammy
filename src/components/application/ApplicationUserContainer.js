import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import ApplicationProgressButtonsWidget from './ApplicationProgressButtonsWidget';
import UserProfileContainer from 'User/UserProfileContainer';
import { startClearApplicationUser, startSetApplicationUser } from 'Actions/application';

export class ApplicationUserContainer extends React.Component {
    constructor(props) {
        super(props);
    }

    handleDecrementStep = () => {
        this.props.startClearApplicationUser();
        this.props.handleDecrementCurrentStep();
    }

    handleIncrementStep = () => {
        this.props.startSetApplicationUser();
        this.props.handleIncrementCurrentStep();
    }

    render() {
        return (
            <div className="application_user_container">
                <ApplicationProgressButtonsWidget
                    currentStep={this.props.currentStep}
                    maxSteps={this.props.maxSteps}
                    handleDecrementCurrentStep={this.handleDecrementStep}
                    handleIncrementCurrentStep={this.handleIncrementStep}
                />
                <UserProfileContainer />
            </div>
        );
    }
}

const mapStateToProps = () => ({
})

const mapDispatchToProps = (dispatch) => ({
    startClearApplicationUser: () => dispatch(startClearApplicationUser()),
    startSetApplicationUser: () => dispatch(startSetApplicationUser()),
})

export default connect(mapStateToProps, mapDispatchToProps)(ApplicationUserContainer);

ApplicationUserContainer.propTypes = {
    currentStep: PropTypes.number.isRequired,
    handleIncrementCurrentStep: PropTypes.func.isRequired,
    handleDecrementCurrentStep: PropTypes.func.isRequired,
    maxSteps: PropTypes.number.isRequired,
    startClearApplicationUser: PropTypes.func.isRequired,
    startSetApplicationUser: PropTypes.func.isRequired,
};
