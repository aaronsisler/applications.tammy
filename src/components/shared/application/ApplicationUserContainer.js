import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import ApplicationProgressButtonsWidget from './ApplicationProgressButtonsWidget';
import UserProfileContainer from '../../user/UserProfileContainer';
import { startClearApplicationUser, startSetApplicationUser } from '../../../actions/application';

export class ApplicationUserContainer extends React.Component {
    constructor(props) {
        super(props);
    }

    handleDecrementCurrentStep = () => {
        this.props.startClearApplicationUser();
        this.props.handleDecrementCurrentStep();
    }

    handleIncrementCurrentStep = () => {
        this.props.startSetApplicationUser();
        this.props.handleIncrementCurrentStep();
    }

    render() {
        return (
            <div className="application_user_container">
                <ApplicationProgressButtonsWidget
                    currentStep={this.props.currentStep}
                    maxSteps={this.props.maxSteps}
                    handleDecrementCurrentStep={this.handleDecrementCurrentStep}
                    handleIncrementCurrentStep={this.handleIncrementCurrentStep}
                />
                <UserProfileContainer />
            </div>
        );
    }
}

const mapStateToProps = () => ({
})

const mapDispatchToProps = (dispatch) => ({
    startClearApplicationUser: () => dispatch(startClearApplicationUser()),
    startSetApplicationUser: () => dispatch(startSetApplicationUser()),
})

export default connect(mapStateToProps, mapDispatchToProps)(ApplicationUserContainer);

ApplicationUserContainer.propTypes = {
    currentStep: PropTypes.number.isRequired,
    maxSteps: PropTypes.number.isRequired,
    handleIncrementCurrentStep: PropTypes.func.isRequired,
    handleDecrementCurrentStep: PropTypes.func.isRequired,
    startClearApplicationUser: PropTypes.func.isRequired,
    startSetApplicationUser: PropTypes.func.isRequired,
};
