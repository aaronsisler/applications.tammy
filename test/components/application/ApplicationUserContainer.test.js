
import React from 'react';
import { shallow } from 'enzyme';
import { ApplicationUserContainer } from 'Application/ApplicationUserContainer';

describe('ApplicationUserContainer', () => {
    const startClearApplicationUser = jest.fn();
    const startDecrementCurrentStep = jest.fn();
    const startIncrementCurrentStep = jest.fn();
    const startSetApplicationUser = jest.fn();
    const ApplicationProgressButtonsWidget = 'Connect(ApplicationProgressButtonsWidget)';
    let wrapper;

    const buildWrapper = () => {
        wrapper = shallow(
            <ApplicationUserContainer
                startClearApplicationUser={startClearApplicationUser}
                startDecrementCurrentStep={startDecrementCurrentStep}
                startIncrementCurrentStep={startIncrementCurrentStep}
                startSetApplicationUser={startSetApplicationUser}
            />);
    };

    beforeEach(() => {
        buildWrapper();
    });

    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    describe(`handleDecrementStep() method on Progress buttons Widget`, () => {
        it('should call startClearApplicationUser()', () => {
            wrapper.find(ApplicationProgressButtonsWidget).prop('handleDecrementStep')();
            expect(startClearApplicationUser).toHaveBeenCalled();
        });

        it('should call startDecrementCurrentStep()', () => {
            wrapper.find(ApplicationProgressButtonsWidget).prop('handleDecrementStep')();
            expect(startClearApplicationUser).toHaveBeenCalled();
        });
    });

    describe(`handleIncrementStep() method on Progress buttons Widget`, () => {
        it('should call startSetApplicationUser()', () => {
            wrapper.find(ApplicationProgressButtonsWidget).prop('handleIncrementStep')();
            expect(startSetApplicationUser).toHaveBeenCalled();
        });

        it('should call startIncrementCurrentStep()', () => {
            wrapper.find(ApplicationProgressButtonsWidget).prop('handleIncrementStep')();
            expect(startIncrementCurrentStep).toHaveBeenCalled();
        });
    });
});
