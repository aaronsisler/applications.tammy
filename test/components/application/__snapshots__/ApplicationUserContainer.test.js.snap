// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationUserContainer renders correctly 1`] = `
<div
  className="application_user_container"
>
  <Connect(ApplicationProgressButtonsWidget)
    handleDecrementStep={[Function]}
    handleIncrementStep={[Function]}
  />
  <Connect(UserProfileContainer) />
</div>
`;
